local/local_test.go

```diff
@@ -51,40 +51,53 @@ func testLocalImage(t *testing.T, when spec.G, it spec.S) {
 	})
 
 	when("#NewImage", func() {
-		when("image is available locally", func() {
-			var repoName = newTestImageName()
+		when("no base image is given", func() {
+			it("returns an empty image", func() {
+				_, err := local.NewImage(newTestImageName(), dockerClient)
+				h.AssertNil(t, err)
+			})
 
-			it.After(func() {
-				h.AssertNil(t, h.DockerRmi(dockerClient, repoName))
+			it("sets sensible defaults for all required fields", func() {
+				// os, architecture, and rootfs are required per https://github.com/opencontainers/image-spec/blob/master/config.md
+				img, err := local.NewImage(newTestImageName(), dockerClient)
+				h.AssertNil(t, err)
+				h.AssertNil(t, img.Save())
+				defer h.DockerRmi(dockerClient, img.Name())
+				inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), img.Name())
+				h.AssertNil(t, err)
+				h.AssertEq(t, inspect.Os, "linux")
+				h.AssertEq(t, inspect.Architecture, "amd64")
+				h.AssertEq(t, inspect.RootFS.Type, "layers")
 			})
+		})
 
-			it("returns the local image", func() {
-				labels := make(map[string]string)
-				labels["some.label"] = "some.value"
+		when("#FromBaseImage", func() {
+			when("base image exists", func() {
+				var repoName = newTestImageName()
 
-				h.CreateImageOnLocal(t, dockerClient, repoName, fmt.Sprintf(`
+				it.After(func() {
+					h.AssertNil(t, h.DockerRmi(dockerClient, repoName))
```

Comment:
It seems strange to have this here without a `Before`, since the image is created in the test. Maybe move it to the end of the test? I suppose it might not get called on a failure though...

I know we discussed the usage of `Before` and `After` previously. In this case, maybe it's best to just put the creation in a `Before` until we have a better pattern for these scenarios.

Reply:
In the test I added I actually did the cleanup in a `defer` statement. I just moved this text into the appropriate context so that the organization would mirror that of of `remote_test.go`